content/ADR-X-quests-poc-user-tracking.md

```diff
@@ -0,0 +1,95 @@
+---
+layout: adr
+adr: X
+title: Quests PoC - User tracking
+date: 2023-03-10
+status: Draft 
+type: RFC 
+spdx-license: CC0-1.0
+authors:
+  - guidota 
+---
+## Abstract
+This document proposes a protobuffer definition and implementation plan for a client-side quests system that can be used in the context of virtual worlds and games. The focus of this document is on user tracking and event processing.
+
+## Context, Reach & Prioritization
+The implementation of a quests system in a virtual world or game can greatly enhance the player experience and provide a sense of achievement and progress. This feature is particularly important for virtual worlds and games that have an open-world structure and rely on player-generated content. A client-side quests system can help guide players towards engaging with the available content and provide incentives for players to create their own content.
+
+## Solution Space Exploration
+
+The client will use the Decentraland RPC and protocol library to communicate with the QuestsService backend. 
+
+```proto
+syntax = "3";
+package decentraland.quests;
+
+message UserAddress {
+  string user_address = 1;
+}
+
+message StartQuestRequest {
+  string user_address = 1;
+  string quest_id = 2;
+}
+
+message StartQuestResponse {
+  bool accepted = 1;
+}
+
+message AbortQuestRequest {
+  string user_address = 1;
+  string quest_instance_id = 2;
+}
+
+message AbortQuestResponse {
+  bool accepted = 1;
+}
+
+message Event {
+  string address = 1;
+  string type = 2; 
+  map<string, string> parameters = 3;
+}
+
+message EventResponse {
+  option fixed32 event_id = 1;
+  bool accepted = 2;
+}
+
+message QuestState {
+  // TODO: single quest state
+}
+
+message UserUpdate {
+  oneof message {
+    repeated QuestState quest_state = 1,
+    fixed32 event_ignored = 2;
+  }
+}
+
+service QuestsService {
+  // User actions
+  rpc StartQuest(StartQuestRequest) returns (StartQuestResponse) {}
+  rpc AbortQuest(AbortQuestRequest) returns (AbortQuestResposne) {}
+  rpc SendEvent(Event) returns (EventResponse) {}
+
+  // Listen to changes in quest states and event processing updates
+  rpc Subscribe(UserAddress) returns (stream UserUpdate) {}
+}
+```
+
+The protobuf definition defines the messages and service necessary for the quests system:
+The `QuestsService` service definition includes three methods that allow users to start and abort quests and send events to the quests system. The `Subscribe` method returns a stream of `UserUpdate` messages that can be used to keep the client-side quests system up to date with the state of the user's quests and event processing.
+
+Once this client is implemented, it would be used to develop a Global Scene in charge of the Quest UI and user tracking logic. Using the SDK UI components is recommended, so it works on any Explorer implementation.
+
+### Quests UI
```

Comment:
Can you also add a link/description of the server implementation?

Reply:
I'd rather not add external links that may be broken or get obsolete in the future. I was thinking that the idea of having a proto service definition is that the client is agnostic to server implementation, what do you say? (Also this document is focused on the client side only)